dandi/pynwb_utils.py

```diff
@@ -362,14 +387,24 @@ def validate(path: Union[str, Path], devel_debug: bool = False) -> List[str]:
         if version is not None:
             # Explicitly sanitize so we collect warnings.
             # TODO: later cast into proper ERRORs
+            # Do we really need to do these acrobatics? string comparison works fine...
             version = _sanitize_nwb_version(version, log=errors.append)
             try:
                 v = semantic_version.Version(version)
             except ValueError:
                 v = None
             if v is not None and v < semantic_version.Version("2.1.0"):
                 errors_ = errors[:]
-                errors = [e for e in errors if not re_ok_prior_210.search(str(e))]
+                errors = []
+                for e in errors:
+                    try:
+                        assert e.message
+                        if not re_ok_prior_210.search(e.message):
+                            errors.append(e)
+                    except AssertionError:
+                        errors.append(e)
```

Comment:
too verbose, I really think there should be some construct like 
```suggestion
                errors = [e for e in errors if not re_ok_prior_210.search(cast(str, getattr(e, "message", ""))]
```
which would keep it concise and please type checking

Reply:
done